### ISSUE TITLE
Flipping screen possible?

### ISSUE THREAD

**Issue by matthicksj** [jakehilborn/displayplacer#33]
I see rotation is possible but can you actually flip the screen horizontally to support a teleprompter?  For example, in xrandr, it would be xrandr -x.  Is this possible with displayplacer?

**Comment by brichards** [comment]
I, too, would love to be able to do this for the very same reason!

**Comment by jakehilborn** [comment]
I don't believe this is possible. I couldn't find anything in the Quartz Display Services documentation. It also doesn't exist in SwitchResX which leads me to believe this doesn't exist.

If you can find something online about being able to flip the screen over the y axis let me know.

**Comment by mwarrenus** [comment]
FWIW, this Apple Stackexchange answer says [SwitchResX includes a horizontal flip](https://apple.stackexchange.com/questions/91174/software-to-horizonally-flip-not-rotate-osxs-monitor-output), but the functionality has been hidden behind an option⌥ click as it's sporadic with new Mac OS versions.

**Comment by mwarrenus** [comment]
Perhaps [kIOScaleInvertX](https://opensource.apple.com/source/IOGraphics/IOGraphics-406/IOGraphicsFamily/IOKit/graphics/IOGraphicsTypes.h.auto.html#:~:text=kIOScaleInvertX) ?

**Comment by garethmarlow** [comment]
I have this exact need (teleprompter). Option-click on SwitchResX brings up the config option but it has no effect. I contacted Stephane, the developer of SwitchResX, who said that he had no way of making this work - it would be brilliant if it could.

**Comment by chrisspiegl** [comment]
I have also found the same answers. SwitchResX had this feature but it's no longer possible in macOS apparently (Stephan said).
I am looking for any resolution that may pop up. But I guess the only real solution would be to use an external monitor that supports flipping in the display firmware.

**Comment by garethmarlow** [comment]
OK I've got a working solution but it's somewhat fiddly.

I'm driving the display I'm using as part of the teleprompter using a raspberry pi, running RPiPlay (https://github.com/FD-/RPiPlay) - it's an Airplay server, pretty stable and minimal latency so it does just feel like an additional direct-connected display. The folks at RPiPlay have recently added flip options - but there's also a boot.txt option on the raspberry pis which will flip the display.

It's been a fun project but it's cost £50 or so to get working, and you need to be reasonably technical to pull it off, so finding a way to do this at the MacOS level would still be great. Meanwhile, video of my setup here: https://www.youtube.com/watch?v=kwrLKgpgZSI&feature=youtu.be

**Comment by chrisspiegl** [comment]
Great setup you have going on there with the wood frame teleprompter. I do think I could pull this off, but I'd rather not want to work with an additional Pi in there. We'll see how it goes 🙈  ‚ maybe someday there will be a solution somehow.
